pkg/generate/app/cmd/newapp.go

```diff
@@ -849,7 +853,7 @@ func (c *AppConfig) Run() (*AppResult, error) {
 		}
 	}
 	// check for circular reference specifically from the newly generated objects, handling new-app vs. new-build nuances as needed
-	err = c.checkCircularReferences(objects)
+	err = c.checkCircularReferences(checkObjects)
```

Comment:
As mentioned earlier, I'm not sure that we need to pass the image stream that we are not creating to the checkCircularReferences call. If the image stream exists, then checkCircularReferences should be able to find it to determine if we indeed have a circular reference.